processing/src/main/java/io/druid/query/aggregation/post/DoubleGreatestPostAggregator.java

```diff
@@ -81,14 +76,12 @@ public Comparator getComparator()
   public Object compute(Map<String, Object> values)
   {
     Iterator<PostAggregator> fieldsIter = fields.iterator();
-    double retVal = Double.NEGATIVE_INFINITY;
-    if (fieldsIter.hasNext()) {
-      retVal = ((Number) fieldsIter.next().compute(values)).doubleValue();
-      while (fieldsIter.hasNext()) {
-        double other = ((Number) fieldsIter.next().compute(values)).doubleValue();
-        if (other > retVal) {
-          retVal = other;
-        }
+    Double retVal = NullHandling.replaceWithDefault() ? Double.NEGATIVE_INFINITY : null;
+    while (fieldsIter.hasNext()) {
+      Number nextVal = ((Number) fieldsIter.next().compute(values));
+      // Ignore NULL values and return the greatest out of non-null values.
+      if (nextVal != null && COMPARATOR.compare(nextVal.doubleValue(), retVal) > 0) {
```

Comment:
This code performs unnecessary unboxing/boxing cycles. I suggest to use `Comparator.comparingDouble(Number::doubleValue)` as `COMPARATOR`, and special-case `retVal == null`

Reply:
done.